src/Orleans.Runtime/GrainDirectory/CachedGrainLocator.cs

```diff
@@ -205,6 +195,25 @@ await foreach (var snapshot in updates)
             }
         }
 
+        private bool IsPointingToDeadSilo(GrainAddress grainAddress)
+            => IsPointingToDeadSilo(grainAddress.SiloAddress, grainAddress.MembershipVersion);
+
+        private bool IsPointingToDeadSilo(SiloAddress siloAddress, MembershipVersion membershipVersion)
```

Comment:
Maybe rename to `IsKnownDeadSilo` to be clear that if this returns `true` then the silo is known to be dead but if it returns false then we do not know whether the silo is dead or not